### PR TITLE
[Merged by Bors] - feat: port Data.Set.Intervals.OrdConnected

### PR DESCRIPTION
**Pull request by LukasMias**

### CODE REVIEW

**Comment by LukasMias** [comment]
I will be gone over the holidays, so if there is still something to be fixed about this file, feel free to implement the changes on my behalf 👋

File: Mathlib/Data/Set/Intervals/OrdConnected.lean [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+theorem OrdConnected_empty : OrdConnected (∅ : Set α) :=
+  ⟨fun _ => False.elim⟩
+#align set.ord_connected_empty Set.OrdConnected_empty
+
+@[instance]
+theorem OrdConnected_univ : OrdConnected (univ : Set α) :=
+  ⟨fun _ _ _ _ => subset_univ _⟩
+#align set.ord_connected_univ Set.OrdConnected_univ
+
+/-- In a dense order `α`, the subtype from an `ord_connected` set is also densely ordered. -/
```
Comment (jcommelin) [inline Mathlib/Data/Set/Intervals/OrdConnected.lean:192]:
```suggestion
/-- In a dense order `α`, the subtype from an `OrdConnected` set is also densely ordered. -/
```

File: Mathlib/Data/Set/Intervals/OrdConnected.lean [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Data.Set.Intervals.UnorderedInterval
+import Mathlib.Data.Set.Lattice
+
+/-!
+# Order-connected sets
+
+We say that a set `s : set α` is `OrdConnected` if for all `x y ∈ s` it includes the
```
Comment (riccardobrasca) [inline Mathlib/Data/Set/Intervals/OrdConnected.lean:17]:
```suggestion
We say that a set `s : Set α` is `OrdConnected` if for all `x y ∈ s` it includes the
```

File: Mathlib/Data/Set/Intervals/OrdConnected.lean [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+
+open OrderDual (toDual ofDual)
+
+namespace Set
+
+section Preorder
+
+variable {α β : Type _} [Preorder α] [Preorder β] {s t : Set α}
+
+/-- We say that a set `s : set α` is `OrdConnected` if for all `x y ∈ s` it includes the
```
Comment (riccardobrasca) [inline Mathlib/Data/Set/Intervals/OrdConnected.lean:37]:
```suggestion
/-- We say that a set `s : Set α` is `OrdConnected` if for all `x y ∈ s` it includes the
```

File: Mathlib/Data/Set/Intervals/OrdConnected.lean [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+interval `[[x, y]]`. If `α` is a `DenselyOrdered` `ConditionallyCompleteLinearOrder` with
+the `OrderTopology`, then this condition is equivalent to `IsPreconnected s`. If `α` is a
+`LinearOrderedField`, then this condition is also equivalent to `Convex α s`. -/
+class OrdConnected (s : Set α) : Prop where
+  -- porting note: added docstring
+  /-- `s : set α` is `OrdConnected` if for all `x y ∈ s` it includes the interval `[[x, y]]`. -/
```
Comment (riccardobrasca) [inline Mathlib/Data/Set/Intervals/OrdConnected.lean:43]:
```suggestion
  /-- `s : Set α` is `OrdConnected` if for all `x y ∈ s` it includes the interval `[[x, y]]`. -/
```

File: Mathlib/Data/Set/Intervals/OrdConnected.lean [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+section Preorder
+
+variable {α β : Type _} [Preorder α] [Preorder β] {s t : Set α}
+
+/-- We say that a set `s : set α` is `OrdConnected` if for all `x y ∈ s` it includes the
+interval `[[x, y]]`. If `α` is a `DenselyOrdered` `ConditionallyCompleteLinearOrder` with
+the `OrderTopology`, then this condition is equivalent to `IsPreconnected s`. If `α` is a
+`LinearOrderedField`, then this condition is also equivalent to `Convex α s`. -/
+class OrdConnected (s : Set α) : Prop where
+  -- porting note: added docstring
```
Comment (riccardobrasca) [inline Mathlib/Data/Set/Intervals/OrdConnected.lean:42]:
```suggestion
```
I don't think this is necessary, it happens everywhere and it is already documented in the wiki.

**Comment by rwbarton** [comment]
bors merge

**Pull request closed without merging**